Feature/recursive hydration 33994095

Bukarinov opened the pull request:
https://www.pivotaltracker.com/story/show/33994095

Bukarinov committed:
Recursive Hydrator

Bukarinov committed:
Tests for Hydrator

Bukarinov committed:
Fields

Bukarinov committed:
Tests for Fields

Bukarinov committed:
Remove Iterator

Bukarinov committed:
Move hydrating data to CreateApiModelListener

Bukarinov committed:
Fix Acl for new module path

Bukarinov committed:
Static hydrators

Bukarinov committed:
Will throw exception if get wrong fields

Bukarinov committed:
Test for wrong fields

Bukarinov committed:
Move Fields to Stdlib/Hydrator

Bukarinov committed:
Move hydrator tests to  GeometriaLabTest/Stdlib/Hydrator

Bukarinov committed:
Rename Hydrator to Extractor

shumkov commented on library/GeometriaLab/Api/View/Http/CreateApiModelListener.php:
```diff
@@ -113,7 +114,8 @@ public function hydrateData(ZendMvcEvent $e)
 
         if ($data instanceof Model\ModelInterface || $data instanceof Model\CollectionInterface) {
-            // $data = hydrate($data);
+            $fields = Fields::createFromString($e->getRequest()->getQuery()->get('_fields'));
```
Переносим эту логику в сервис

shumkov commented on library/GeometriaLab/Stdlib/Extractor/Extractor.php:
```diff
@@ -0,0 +1,119 @@
+    /**
+     * @var Fields
+     */
+    protected $fields;
```
Зачем?

shumkov commented on library/GeometriaLab/Stdlib/Extractor/Extractor.php:
```diff
@@ -0,0 +1,119 @@
+                continue;
+            }
+            if (is_subclass_of($source, get_class())) {
+                /* @var Extractor $extractor */
```
Почему не из сервис менеджера?

shumkov commented on library/GeometriaLab/Stdlib/Extractor/Fields.php:
```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace GeometriaLab\Stdlib\Extractor;
```
Используем обычный массив

shumkov commented on library/GeometriaLab/Stdlib/Extractor/Service.php:
```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace GeometriaLab\Stdlib\Extractor;
```
Перенести в Api

Bukarinov committed:
Remove Fields.
Move ExtractorService from Stdlib\Extractor\Service to Api\Stdlib\Extractor\Service.

Bukarinov committed:
Remove Fields again.

Bukarinov committed:
Namespace for Extractor

Bukarinov committed:
Acl configuration

Bukarinov committed:
Move Extractor from Stdlib to Api/Stdlib
